server/api/user/update/controller.js

```diff
@@ -0,0 +1,156 @@
+import Joi from 'joi';
+import Boom from 'boom';
+import compose from 'koa-compose';
+import isNull from 'lodash/isNull';
+import packJSONRPC from '../../../middleware/packJSONRPC';
+import { validateRequest } from '../../../middleware/validation';
+import {
+  decorateSession,
+  isUserAuthenticated,
+  decorateUserPermissions,
+  findUserPermissionIndex,
+} from '../../../middleware/auth';
+import updateUser from './service';
+import getUserInfo from '../info/service';
+import { AuthorizationError } from '../../../constants/errors';
+
+export const validationSchema = {
+  body: {
+    id: Joi.string()
+      .length(24)
+      .hex()
+      .required(),
+    username: Joi.string()
+      .lowercase()
+      .trim()
+      .min(2)
+      .max(30)
+      .optional()
+      .regex(/^[A-Za-z0-9_\-.]*$/, { name: 'username' }),
+    password: Joi.string()
+      .trim()
+      .min(8)
+      .max(50)
+      .optional(),
+    fullName: Joi.string()
+      .trim()
+      .min(2)
+      .max(100)
+      .optional(),
+    picture: Joi.string()
+      .allow(null)
+      .uri({ scheme: ['http', 'https'] })
+      .trim()
+      .max(500)
+      .optional(),
+    emails: Joi.array()
+      .items(
+        Joi.object()
+          .unknown(false)
+          .keys({
+            address: Joi.string()
+              .trim()
+              .email()
+              .max(100)
+              .required(),
+            isVerified: Joi.any()
+              .forbidden(),
+            isPrimary: Joi.boolean()
+              .default(false)
+              .optional(),
+          })
+          .required()
+      )
+      .min(1)
+      .max(10)
+      .unique((a, b) => a.address === b.address)
+      .optional(),
+  },
+};
+
+const isRequestorAllowedToEditUser = (requestorPermissions, orgId) => {
+  const hasUserReadPermissions =
+    findUserPermissionIndex(requestorPermissions, {
+      name: 'yeep.user.Write',
```

Comment:
Oops, I believe name is case sensitive. In any case please change this to `yeep.user.write` (all lowercase).